### PR TITLE
INFRA-76: Add reusable workflow for deployments

### PR DESCRIPTION
**Pull request by pyaqubzade**
## Added 

- **deploy** **reusable** action for
  - Deploying **Java** applications with **Gradle** to Kubernetes cluster

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+          docker build -t $ECR_REGISTRY/$ECR_REPOSITORY:$IMAGE_TAG .
+          docker push $ECR_REGISTRY/$ECR_REPOSITORY:$IMAGE_TAG
+          echo "::set-output name=image::$ECR_REGISTRY/$ECR_REPOSITORY:$IMAGE_TAG"
+
+      - name: Helm template
+        env:
+          ECR_REGISTRY: ${{ steps.login-ecr.outputs.registry }}
+          ECR_REPOSITORY: ${{ github.event.repository.name }}
+          IMAGE_TAG: ${{ github.sha }}
+          ENV: ${{ inputs.env }}
```
Comment (pyaqubzade) [inline .github/workflows/deploy.yml:126]: You forgot to request env input from caller

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -0,0 +1,133 @@
+on:
+  workflow_call:
+    inputs:
+      environment:
```
Comment (GGurbanov) [inline .github/workflows/deploy.yml:4]: I would suggest to add `description` as well

File: .github/workflows/deploy.yml [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+        required: true
+        type: string
+      platform-url:
```
Comment (GGurbanov) [inline .github/workflows/deploy.yml:7]: The same here

File: .github/workflows/deploy.yml [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+        type: string
+      gradle-version:
+        description: 'Gradle version'
+        required: false
+        default: 'wrapper'
+        type: string
+      gradle-debug-cache:
+        description: 'Debug output for gradle cache'
+        required: false
+        default: 'true'
```
Comment (GGurbanov) [inline .github/workflows/deploy.yml:29]: Better to keep boolean value without quotes

File: .github/workflows/deploy.yml [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+
+      - name: Apply Kube configuration
+        env:
+          KUBE_CONFIG_DATA: ${{ secrets.kube-config }}
+        run: |
+          mkdir ~/.kube
+          echo $KUBE_CONFIG_DATA | base64 --decode > ~/.kube/config
+
+      - name: Install AWS IAM
+        uses: woofenator/action-aws-iam-authenticator@master
```
Comment (GGurbanov) [inline .github/workflows/deploy.yml:85]: Always prefer to use a specific version rather than branch name

**Review by GGurbanov — APPROVED** [review]
LGTM, my comments aren't blocker for this pr

**Pull request merged**